File: 240325/처음 등장하는 위치/first-appearing-position.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] numbers = new int[n];
        for (int i = 0; i < n; i++) {
            numbers[i] = sc.nextInt();
        }

        Map<Integer, Integer> map = new TreeMap<>();
        int index = 1;
        for (int number : numbers) {
            if (map.containsKey(number)) {
                continue;
            }
            map.put(number, index);
            index++;
        }

        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            System.out.printf("%d %d\n", entry.getKey(), entry.getValue());
        }
    }
}

File: 240325/처음 등장하는 위치/first-appearing-position.kt
import java.util.*

fun main() {
    val n = readLine()!!.toInt()
    val numbers = readLine()!!.split(" ").map { x -> x.toInt() }.toList()

    val map: TreeMap<Int, Int> = TreeMap()

    var index = 1
    for (number in numbers) {
        if (!map.containsKey(number)) {
            map[number] = index
        }
        index++
    }

    for ((number, index) in map.entries) {
        println("$number $index")
    }
}